### PR TITLE
Dev docker config

### PR DESCRIPTION
**Pull request by thomascube**
* Generate variants from templates
  This is a slightly modified version of PR #55 and fixes issue #5.
* Write individual config file from env vars
  This implements the changes proposed in #112

### CODE REVIEW

**Comment by thomascube** [comment]
@J0WI @madmath03 I'd like another pair of eyes on these changes and therefore kindly inviting you to review this. Merci!

File: generate.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	[fpm-alpine]='alpine'
+)
+
+declare -A EXTRAS=(
+	[apache]='¬RUN a2enmod rewrite'
+	[fpm]=''
+	[fpm-alpine]=''
+)
+
+VERSION="${1:-$(curl -sS https://roundcube.net/VERSION.txt)}"
```
Comment (J0WI) [inline generate.sh:22]:
```suggestion
VERSION="${1:-$(curl -fsS https://roundcube.net/VERSION.txt)}"
```
so curl fails on HTTP errors.

**Review by J0WI — COMMENTED** [review]
@docker-library-bot would expect a file named `update.sh` (https://github.com/roundcube/roundcubemail-docker/issues/5#issuecomment-395901195). 

https://github.com/roundcube/roundcubemail-docker/pull/55 also adds a `generate-stackbrew-library.sh` for https://github.com/roundcube/roundcubemail-docker/issues/31 though this is not a requirement.

**Comment by thomascube** [comment]
> #55 also adds a `generate-stackbrew-library.sh` for #31 though this is not a requirement.

Yeah, I separated that from the current change because I first want to understand the processes required for official images. So let's take one step after the other.

**Review by J0WI — APPROVED** [review]

**Pull request merged**